### ISSUE TITLE
lxd depends on the network to launch cached images

### ISSUE THREAD

**Issue by niemeyer** [lxc/lxd#2602]
Even for images lxd already has on cache, the remote servers are being consulted. This was the outcome of trying to use lxd while on a plane this weekend (Ubuntu 16.04):

```
Creating ubuntu-16-04-64
error: Get https://cloud-images.ubuntu.com/releases/streams/v1/index.json: lookup cloud-images.ubuntu.com on 127.0.1.1:53: server misbehaving
```

and:

```
Creating ubuntu-16-04-64
error: Get https://images.linuxcontainers.org/streams/v1/index.json: lookup images.linuxcontainers.org on 127.0.1.1:53: server misbehaving
```

This apparently also means that when the remote image is updated, lxd downloads the new image instead of using the local cached one. This already caused me issues while attempting to use lxd on stage, as the logic would both go to the network, and also _use the network heavily_ instead of using its local cache.

Besides those sporadic issues, this is also quite relevant on production use cases, as it turns these remote servers into single points of failure.

**Comment by stgraber** [comment]
LXD 2.5 maintains its stream cache on disk rather than in memory, making such offline operation possible.
That commit will be part of LXD 2.0.6 so will be included in 16.04 "soon".

As for the fact that LXD will always grab the latest version of the image, that's done by design to ensure fast roll out of security fixes to users. That being said, we only actually hit the simplestreams server up to once per hour. If you want to start the particular image you have in your local cache, use its fingerprint (or one of its aliases if you have one set).

Marking as closed as this is effectively a duplicate of #2487

**Comment by niemeyer** [comment]
Thanks, that solves most of the problems. The only unfortunate detail left is the fact it will block a fast deployment of a container at an arbitrary point. This means production uses will need to implement fingerprint mapping locally if they don't want to pay the cost of an image update in flight.

It'd be nice to have a `--cached` flag or similar to prevent that behavior.
